### DISCUSSION TITLE
ZKsync doesn't solve the blockchain trilemma

### DISCUSSION THREAD

**Discussion by LordAdhaar** [Cyfrin/foundry-full-course-cu#2298] [Q&A]
I am probably wrong, correct me.

The blockchain trilemma suggests that a blockchain cannot be fully decentralized, secure, and scalable all at once. ZKsync aims to address Ethereum's scalability challenge using ZK Rollups.

Currently, ZKsync is in its early stages (often referred to as "stage 0"), meaning that while it improves scalability, it remains somewhat centralized. This centralization, in turn, partially compromises the decentralization aspect, illustrating the ongoing challenge of achieving all three pillars of the trilemma simultaneously.

**Comment by EngrPips** [comment]
It seems like you get the gist.

**Reply by LordAdhaar** [reply]
Did I miss something? Am I wrong or right?

**Reply by EngrPips** [reply]
You are correct as far as my knowledge encompasses.

**Reply by LordAdhaar** [reply]
Which part of the course are you on?

**Reply by EngrPips** [reply]
I finished the whole course already, but I am currently re-taking the `EVM AND FORMAL VERIFICATION` course so as to properly document my codebase.

**Reply by JusticeSol** [reply]
@LordAdhaar I think your explanation on the trilemma topic is right

**Reply by LordAdhaar** [reply]
I could find only one stage 3 L2

**Reply by LordAdhaar** [reply]
> I finished the whole course already, but I am currently re-taking the `EVM AND FORMAL VERIFICATION` course so as to properly document my codebase.

Are you now going for auditing competitions ?

**Reply by EngrPips** [reply]
I have done a few ones before but didn't make money but I am looking to getting back into auditing soon.
